client/src/components/bpmn.modeler.component.jsx

```diff
@@ -1,69 +1,104 @@
+      moddleExtensions: {
+        camunda: camundaModdleDescriptor,
+      },
+    });
 
-    componentDidMount = () => {
-        this.modeler = new BpmnModeler({
-            container: "#bpmnview",
-            keyboard: {
-                bindTo: window,
-            },
-            propertiesPanel: {
-                parent: "#propview",
-            },
-            additionalModules: [propertiesPanelModule, propertiesProviderModule],
-            moddleExtensions: {
-                camunda: camundaModdleDescriptor,
-            },
-        });
+    this.newBpmnDiagram();
+  };
 
-        this.newBpmnDiagram();
-    };
+  newBpmnDiagram = () => {
+    this.openBpmnDiagram(emptyBpmn);
+  };
 
-    newBpmnDiagram = () => {
-        this.openBpmnDiagram(emptyBpmn);
-    };
+  openBpmnDiagram = (xml) => {
+    this.modeler.importXML(xml, (error) => {
+      if (error) {
+        return console.log('fail import xml');
+      }
 
-    openBpmnDiagram = (xml) => {
-        this.modeler.importXML(xml, (error) => {
-            if (error) {
-                return console.log("fail import xml");
-            }
+      var canvas = this.modeler.get('canvas');
 
-            var canvas = this.modeler.get("canvas");
+      canvas.zoom('fit-viewport');
+    });
+  };
 
-            canvas.zoom("fit-viewport");
-        });
-    };
+  downloadString = (text, fileType, fileName) => {
+    var blob = new Blob([text], { type: fileType });
 
-    render = () => {
-        return (
-            <div id="bpmncontainer">
-                <div
-                    id="propview"
-                    style={{
-                        width: "25%",
-                        height: "98vh",
-                        float: "right",
-                        maxHeight: "98vh",
-                        overflowX: "auto",
-                    }}
-                ></div>
-                <div
-                    id="bpmnview"
-                    style={{width: "75%", height: "98vh", float: "left"}}
-                ></div>
-            </div>
-        );
-    };
+    var a = document.createElement('a');
+    a.download = fileName;
+    a.href = URL.createObjectURL(blob);
+    a.dataset.downloadurl = [fileType, a.download, a.href].join(':');
+    a.style.display = 'none';
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    setTimeout(function () {
+      URL.revokeObjectURL(a.href);
+    }, 1500);
+  };
+
+  getBpmnDiagramXML = () => {
+    this.modeler.saveXML().then((json) => {
+      console.log(json.xml);
+      var xml = json.xml;
+      this.downloadRobotFile(xml);
+    });
+  };
+
+  downloadRobotFile = (xml) => {
+    var body = convert.xml2json(xml, { compact: true, spaces: 4 });
+    let jsonBody = JSON.parse(body);
+    console.log(jsonBody['bpmn2:definitions']);
+    let robot = parser.parseDiagramJson(jsonBody);
+    this.downloadString(robot, 'text/robot', 'testRobot.robot');
```

Comment:
Is the filename hardcoded or is there any chance to allow for renaming of the efile before downloading? If it is out of scope for now maybe create a new issue for the future?

Reply:
It is hardcoded and I think renaming is out of scope for now. Sandro just created a new Issue for that.

Reply:
Good point though.